Streetcode/Streetcode.XUnitTest/MediatRTests/StreetCode/RelatedTerm/Delete/DeleteRelatedTermHandlerTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿using AutoMapper;
+using FluentResults;
+using MediatR;
+using Microsoft.EntityFrameworkCore.Query;
+using Moq;
+using Streetcode.BLL.DTO.Streetcode.TextContent;
+using Streetcode.BLL.MediatR.Streetcode.RelatedTerm.Delete;
+using Streetcode.DAL.Entities.Streetcode.TextContent;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using Entity = Streetcode.DAL.Entities.Streetcode.TextContent.RelatedTerm;
+
+namespace Streetcode.XUnitTest.MediatRTests.StreetCode.RelatedTerm.Delete
+{
+    public class DeleteRelatedTermHandlerTests
+    {
+        private readonly Mock<IRepositoryWrapper> _repositoryWrapperMock;
+        private readonly Mock<IMapper> _mapperMock;
+
+        public DeleteRelatedTermHandlerTests()
+        {
+            _repositoryWrapperMock = new Mock<IRepositoryWrapper>();
+            _mapperMock = new Mock<IMapper>();
+        }
+
+        [Theory]
+        [InlineData(1, "test", 1)]
+        public async void Handle_WhenRelatedTermExists_DeletesItAndReturnsSuccessResult(int id, string word, int termId)
+        {
+            // Arrange
+            var relatedTerm = new Entity { Id = id, Word = word, TermId = termId };
+            _repositoryWrapperMock.Setup(r => r.RelatedTermRepository.GetFirstOrDefaultAsync(It.IsAny<Expression<Func<Entity, bool>>>(),
+                It.IsAny<Func<IQueryable<Entity>, IIncludableQueryable<Entity, object>>>())).ReturnsAsync(relatedTerm);
+            _repositoryWrapperMock.Setup(r => r.RelatedTermRepository.Delete(relatedTerm));
+            _repositoryWrapperMock.Setup(x => x.SaveChangesAsync()).ReturnsAsync(1);
+
+            var handler = new DeleteRelatedTermHandler(_repositoryWrapperMock.Object, _mapperMock.Object);
+            var command = new DeleteRelatedTermCommand(id);
+
```

Comment:
do we need a command variable?

Reply:
Yes, we do. We pass it to the handler.Handle.